src/js/utils/html-utils.js

```diff
@@ -6,41 +6,78 @@
+         * The intention of an HTML snippet is to capture a string that
+         * is known to contain HTML that has been safely escaped.
+         * As an example, this allows interpolate to understand that
+         * it does not need to further escape this HTML.
          *
          * @param {string} htmlString The string of HTML.
-         * @returns {string} A string flagged as an HTML snippet.
+         * @constructor
+         */
+        function HtmlSnippet(htmlString) {
+            this.text = htmlString;
+        }
+        HtmlSnippet.prototype.valueOf = function() {
+            return this.text;
+        };
+        HtmlSnippet.prototype.toString = function() {
+            return this.text;
+        };
+
+        /**
+         * Helper function to create an HTML snippet from a string.
+         *
+         * @param {string} htmlString The string of HTML.
+         * @returns {HtmlSnippet} An HTML snippet that can be safely rendered.
          * @constructor
          */
         HTML = function(htmlString) {
-            var htmlSnippet = new String(htmlString); // jshint ignore:line
-            htmlSnippet.isHtmlSnippet = true;
-            return htmlSnippet;
+            return new HtmlSnippet(htmlString);
         };
 
         /**
-         * Returns an HTML string that is appropriately escaped.
+         * Ensures that the specified HTML is encoded as an HtmlSnippet.
          *
-         * Note: if the string provided is flagged as being an HTML snippet
+         * If the HTML is provided as a string, then it is HTML-escaped and
+         * returned as an HtmlSnippet. If the parameter is an HTML snippet
          * then it will be returned directly so as not to double escape it.
          *
-         * @param {string} text A string that is either plan text, or is
-         * flagged as being an HTML snippet.
-         * @returns {string} A safely HTML-escaped string.
+         * @param {(string|HtmlSnippet)} html HTML provided either as
+         * a string or as an HtmlSnippet.
+         * @returns {HtmlSnippet} A safely escaped HTML snippet.
          */
-        escape = function(text) {
-            if (text.isHtmlSnippet) {
-                return text.toString();
+        ensureHtml = function(html) {
+            if (html instanceof HtmlSnippet) {
+                return html;
             } else {
-                return _.escape(text);
+                return HTML(_.escape(html));
             }
         };
 
         /**
+         * A wrapper for $ that returns a new JQuery element for the specified HTML.
+         *
+         * Note that the HTML must have a single root element which will be the
+         * one returned. An error will be thrown if the HTML cannot be interpreted
+         * as a single root element.
+         *
+         * @param {HtmlSnippet} html The desired HTML provided as an HtmlSnippet.
+         * @returns {element} A new JQuery element.
+         */
+        element = function(html) {
```

Comment:
I've decided to remove this method for now so we can get this PR merged.

I was thinking we'd need it for calls to ``appendTo`` and ``prependTo`` where the caller has to create a new element from some HTML before attaching it. Here's one example:

https://github.com/edx/edx-platform/blob/master/lms/templates/courseware/progress_graph.js#L9

I'll revisit it when I have some more concrete examples where the existing methods won't be enough.